src/pudl/scripts/dbt_helper.py

```diff
@@ -365,12 +374,26 @@ def update_row_counts(
 
     partition_expr = partition_expressions[0]  # TODO: support multiple partitions
     new = _calculate_row_counts(table_name, partition_expr)
+
+    # Make old and new row counts comparable so we can detect changes
+    row_count_idx = ["table_name", "partition"]
+    if (
+        old.sort_values(by=row_count_idx)
+        .reset_index(drop=True)
+        .equals(new.sort_values(by=row_count_idx).reset_index(drop=True))
```

Comment:
`df.equals()` has less flexibility in terms of what you ask it to check than `assert_frame_equal` so switching to this method means more dataframe munging upstream.